LibreNMS/OS/ArubaosCx.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/*
+ * ArubaosCx.php
+ *
+ * NAC polling including 802.1x and device-profile entries.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       https://www.librenms.org
+ */
+
+namespace LibreNMS\OS;
+
+use App\Models\PortsNac;
+use Illuminate\Support\Collection;
+use LibreNMS\Interfaces\Polling\NacPolling;
+use SnmpQuery;
+
+class ArubaosCx extends \LibreNMS\OS implements NacPolling
+{
+    public function pollNac()
+    {
+        $nac = new Collection();
+
+        $rowSet = [];
+        $ifIndex_map = $this->getDevice()->ports()->pluck('port_id', 'ifName');
+        $table = SnmpQuery::mibDir('arubaos-cx')->mibs(['ARUBAWIRED-PORT-ACCESS-MIB'])->hideMib()->enumStrings()->walk('arubaWiredPortAccessClientTable')->table(2);
+
+        foreach ($table as $ifIndex => $entry) {
+            foreach ($entry as $macKey => $macEntry) {
+                $rowSet[$macKey] = [
+                    'domain' => '',
+                    'ip_address' => '',
+                    'host_mode' => '',
+                    'authz_by' => '',
+                    'username' => '',
+                    'timeout' => '',
+                ];
+                $rowSet[$macKey]['authc_status'] = $macEntry['arubaWiredPacAuthState'];
```

Comment:
Would ` ?? '';` be acceptable as well?  That way if a field does not have a value it would present as empty instead of a string?

Reply:
> In case a switch would not reply all the entries it is supposed to reply, it would be safer to write it like this : `$rowSet[$macKey]['authc_status'] = $macEntry['arubaWiredPacAuthState'] ?? 'unknown';` So the default value of 'unknown' would be displayed, instead of crashing the poller :) This is valid for all `$macEntry[xxxxxxxxx]` calls.

After looking at some of the other examples and considering the options, since some of the values could be empty on purpose I went with `?? ''`.  I didn't want to populate fields with `unknown` if they were supposed to be blank. For the vlan field, since it is an integer in the db, I moved it to `null` instead.

Reply:
Fine with the ` ?? '' `. Whatever fits best to the GUI. And `null` if the DB field is nullable.